prime-router/src/main/kotlin/fhirengine/utils/HL7MessageHelpers.kt

```diff
@@ -127,32 +131,34 @@ object HL7MessageHelpers : Logging {
                 null
             } else Terser(messages[0])
         } else null
-
+        val time = DTM(null)
+        time.setValue(Date())
         // The extraction of these values mimics how the COVID HL7 serializer works
-        var sendingApp = firstMessage?.get("MSH-3-1")
-        if (sendingApp.isNullOrBlank()) sendingApp = "CDC PRIME - Atlanta"
+        var sendingApp =
+            "${firstMessage?.get("MSH-3-1") ?: REPORT_STREAM_APPLICATION_NAME}^" +
```

Comment:
Do we do this kind of code anywhere else, where we build a HL7 string? Should we be using a library/class/method for this instead? I don't like that we are building a HL7 string in this random function. at the very least "getSendingApp" should be a method somewhere with its own unit tests? Thoughts?

Reply:
Yeah, it seems like we do this a lot in the Covid Pipeline. It looks like this was copied from `createHeaders` in `HL7Serializer.kt`. We tried to get away from doing that using the FHIR to HL7 schemas, but there doesn't seem to be a way to do it for the batch headers.
We can write a separate ticket to refactor the code and add better unit tests. This just needed to be fixed right now to be able to send a message to CA.